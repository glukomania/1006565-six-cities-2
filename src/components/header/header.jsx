import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

const Header = (props) => {

  return <header className="header">
    <div className="container">
      <div className="header__wrapper">
        <div className="header__left">
          <Link className="header__logo-link header__logo-link--active" to="/">
            <img className="header__logo" src={props.isInner ? `../img/logo.svg` : `img/logo.svg`} alt="6 cities logo" width="81" height="41" />
          </Link>
        </div>
        <nav className="header__nav">
          <ul className="header__nav-list">
            <li className="header__nav-item user">
              <Link className="header__nav-link header__nav-link--profile" to={props.isAuthorized ? `/favorites` : `/login`}>
                <div className="header__avatar-wrapper user__avatar-wrapper">
                </div>
                <span className="header__user-name user__name">{props.isAuthorized === false ? `Sign in` : props.userCredentials.email}</span>
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>;
};

Header.propTypes = {
  isAuthorized: PropTypes.bool.isRequired,
  userCredentials: PropTypes.object,
  isInner: PropTypes.bool.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isAuthorized: state.isAuthorized,
  userCredentials: state.userCredentials,
});

const mapDispatchToProps = {
};

export {Header};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
