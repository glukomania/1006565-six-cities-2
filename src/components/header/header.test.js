import renderer from 'react-test-renderer';
import {Header} from './header';

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`Header is displayed correctly for unauthorized users`, () => {

  const props = {
    isInner: true,
    isAuthorized: false
  };

  const header = renderer
    .create(<Header {...props} />)
  .toJSON();

  expect(header).toMatchSnapshot();
});


it(`Header is displayed correctly for authorized users`, () => {

  const props = {
    isInner: true,
    isAuthorized: true,
    userCredentials: {email: `test@test.com`}
  };

  const header = renderer
    .create(<Header {...props} />)
  .toJSON();

  expect(header).toMatchSnapshot();
});
