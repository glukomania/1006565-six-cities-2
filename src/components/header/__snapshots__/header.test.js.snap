// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is displayed correctly for authorized users 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      />
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          />
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header is displayed correctly for unauthorized users 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      />
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          />
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
